code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,117 @@
+// ***************************************
+// *********** Web Spit
+// ***************************************
+/datum/action/xeno_action/activable/web_spit
+	name = "Web Spit"
+	ability_name = "Web Spit"
+	mechanics_text = "We spit a stretchy web at our prey"
+	action_icon_state = "toggle_bomb0" // temporary until I get my own icons
+	plasma_cost = 1
+	cooldown_timer = 5 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_WEB_SPIT
+
+/datum/action/xeno_action/activable/web_spit/use_ability(atom/target)
```

Comment:
general remark, you need to call succeed_activate() and add_cooldown() when an ability proc